### PR TITLE
Bit_Trie  Implementation for Max Xor Computation in an Array

### PR DESCRIPTION
**Pull request by DataWorshipper**
#### Description of Change

This pull request adds a **C++ implementation of the Bitwise Trie (Binary Trie)** to solve the **Maximum XOR of Two Numbers in an Array** problem from LeetCode: [Maximum XOR Problem](https://leetcode.com/problems/maximum-xor-of-two-numbers-in-an-array/).

The algorithm constructs a binary trie for all 64-bit integers in the array and efficiently queries the maximum XOR achievable for each number. The implementation is self-contained, includes comprehensive Doxygen-style documentation, and comes with multiple test cases to ensure correctness.

---

#### Checklist

- [x] Added description of change  
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines) (`max_xor_bit_trie.cpp`)  
- [x] Added tests and example, test must pass  
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)  
- [x] Relevant documentation/comments is changed or added  
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)  
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.  
- [x] I acknowledge that all my contributions will be made under the project's license.  

**Notes:** Implements an efficient **O(n * log(MAX_VAL))** solution for finding the maximum XOR pair in an array using a bitwise trie. Includes extensive tests for edge cases, duplicates, and large arrays.

### CODE REVIEW

File: bit_manipulation/max_xor_bit_trie.cpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+ * (https://leetcode.com/problems/maximum-xor-of-two-numbers-in-an-array/)
+ *
+ * @details
+ * Given an array of n integers, the task is to find the maximum XOR value
+ * obtainable by XOR-ing any two numbers in the array. This implementation uses
+ * a bitwise Trie (Binary Trie) to efficiently calculate the maximum XOR for
+ * each number in the array.
+ *
+ * Worst Case Time Complexity: O(n * log(MAX_VAL)) where MAX_VAL is the maximum
+ * value in the array (64-bit integers here) Space Complexity: O(n *
```
Comment (realstealthninja) [inline bit_manipulation/max_xor_bit_trie.cpp]: Line break for the space complexity

File: bit_manipulation/max_xor_bit_trie.cpp [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        return answer;
+    }
+};
+
+/**
+ * @brief Compute the maximum XOR of any two numbers in the array
+ * @param nums vector of unsigned 64-bit integers
+ * @return maximum XOR of any pair
+ */
+std::uint64_t findMaximumXOR(const std::vector<std::uint64_t>& nums) {
```
Comment (realstealthninja) [inline bit_manipulation/max_xor_bit_trie.cpp]: Follow snake case.

File: bit_manipulation/max_xor_bit_trie.cpp [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+}  // namespace max_xor_bit_trie
+}  // namespace bit_manipulation
+
+/**
+ * @brief Self-test implementations
+ */
+static void test() {
+    using bit_manipulation::max_xor_bit_trie::findMaximumXOR;
+
```
Comment (realstealthninja) [inline bit_manipulation/max_xor_bit_trie.cpp]: Add empty vector test.

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Comment by DataWorshipper** [comment]
Hello i have commited the changes that you requested for also followed snake case conventions,could you please review the changes
